tests/test_visitors/test_ast/test_statements/test_variable_used_outside_of_block.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    VariableUsedOutsideOfForBlockViolation,
+    VariableUsedOutsideOfWithBlockViolation
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    VariableUsedOutsideOfBlockVisitor,
+)
+
+variable_used_outside_with = """
+def function():
+    with a.open() as x:
+        some()
+        function()
+        calls()
+    print("This is not allowed".format(x))
+"""
+multiple_variables_defined_by_with = """
+with a.open(), b.open() as c, d.open() as (e, f):
+    print("Hello")
+print(f)
+"""
+
+variable_used_outside_for = """
+for i in range(100):
+    print(i)
+sum = i + 1
+"""
+
+multiple_variables_defined_by_for = """
+for (a, b, c, d) in enumerate(enumerate(enumerate(range(1)))):
```

Comment:
`enumerate(enumerate(enumerate(range(1))))` can be reduces to just `some_call()`